acnportal/acnsim/simulator.py

```diff
@@ -210,6 +227,170 @@ def index_of_evse(self, station_id):
             raise KeyError("EVSE {0} not found in network.".format(station_id))
         return self.network.station_ids.index(station_id)
 
+
+    def to_dict(self, context_dict=None):
+        """
+        Implements BaseSimObj.to_dict. Certain simulator attributes are
+        not serialized completely as they are not ACN-Sim objects
+        (signals and scheduler exist in their own modules).
+
+        If the Python version used is less than 3.7, datetimes cannot be
+        accurately loaded. As such, a warning is thrown when the start
+        attribute is serialized.
+
+        The signals attribute is only serialized if it is natively
+        JSON Serializable, otherwise None is stored.
+
+        Only the scheduler's name is serialized.
+        """
+        context_dict, = base.none_to_empty_dict(context_dict)
+        args_dict = {}
+
+        # Serialize non-nested attributes.
+        nn_attr_lst = [
+            'period', 'max_recompute', 'verbose', 'peak',
+            '_iteration', '_resolve', '_last_schedule_update',
+            'schedule_history'
+        ]
+        for attr in nn_attr_lst:
+            args_dict[attr] = getattr(self, attr)
+
+        args_dict['network'] = \
+            self.network.to_registry(context_dict=context_dict)['id']
+
+        args_dict['scheduler'] = f'{self.scheduler.__module__}' \
+                                 f'.{self.scheduler.__class__.__name__}'
+
+        args_dict['event_queue'] = \
+            self.event_queue.to_registry(context_dict=context_dict)['id']
+
+        if sys.version_info[1] < 7:
+            warnings.warn(f"Datetime {self.start} will not be loaded as "
```

Comment:
Use strftime and strptime or timestamp so it is compatible with versions prior to 3.7